pkg/graveler/graveler.go

```diff
@@ -1314,7 +1314,7 @@ func (g *Graveler) monitorRetries(ctx context.Context, retries int, repositoryID
 // information on the algorithm used.
 func (g *Graveler) prepareForCommitIDUpdate(ctx context.Context, repository *RepositoryRecord, branchID BranchID, operation string) error {
 	return g.retryBranchUpdate(ctx, repository, branchID, func(currBranch *Branch) (*Branch, error) {
-		empty, err := g.isStagingEmpty(ctx, repository, currBranch)
+		empty, err := g.isStagingEmpty(ctx, repository, branchID)
```

Comment:
It's subtle, but notice that this function is being called from somewhere else. So I believe this should be passed:
```suggestion
		empty, err := g.isStagingEmpty(ctx, repository, currBranch.BranchID)
```